src/transaction.ts

```diff
@@ -436,26 +439,34 @@ export class Snapshot extends EventEmitter {
       addLeaderAwareRoutingHeader(headers);
     }
 
-    this.request(
-      {
-        client: 'SpannerClient',
-        method: 'beginTransaction',
-        reqOpts,
-        gaxOpts,
-        headers: headers,
-      },
-      (
-        err: null | grpc.ServiceError,
-        resp: spannerClient.spanner.v1.ITransaction
-      ) => {
-        if (err) {
-          callback!(err, resp);
-          return;
+    const q = {opts: this._observabilityOptions};
```

Comment:
> sometimes we have newly injected values of
When will we be having that. IMU this can only be set at the time of spanner creation via spanner options

If we define the basic object in constructor then we can extend that object whereever we need using "Object.assign" . WDYT ?

Reply:
I'd say let's keep it as is because as I mentioned there are instances where we copy over _observabilityOptions to an already existing object so keeping it in the constructor is non-kosher.